File: code/java/isSubsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if(s.length() == 0)
            return true;
        int i = 0;
        int j = 0;
        while(j < t.length()){
            if(s.charAt(i) == t.charAt(j))
                i = i + 1;
            j = j + 1;
            if(i == s.length())
                return true;
        }
        return false;
    }
}

File: code/python/isSubsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if s == "":
            return True
        i = 0
        j = 0
        while j < len(t):
            if s[i] == t[j]:
                i = i + 1
            j = j + 1
            if i == len(s):
                return True
        return False
        
